connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -403,7 +403,7 @@ private void convertMessages(ConsumerRecords<byte[], byte[]> msgs) {
                     keyAndSchema.schema(), keyAndSchema.value(),
                     valueAndSchema.schema(), valueAndSchema.value(),
                     msg.offset(),
-                    (msg.timestampType() == TimestampType.NO_TIMESTAMP_TYPE) ? null : msg.timestamp(),
+                    (msg.timestamp() <= Record.NO_TIMESTAMP) ? null : msg.timestamp(),
```

Comment:
Yeah that makes sense, that was me misinterpreting what it means to have a negative value other than -1. 

I'll throw an exception instead if that works